pyresample/utils.py

```diff
@@ -423,22 +423,33 @@ def proj4_radius_parameters(proj4_dict):
         new_info = proj4_str_to_dict(proj4_dict)
     else:
         new_info = proj4_dict.copy()
-
+      
     # load information from PROJ.4 about the ellipsis if possible
-    if '+a' not in new_info or '+b' not in new_info:
-        import pyproj
-        ellps = pyproj.pj_ellps[new_info.get('+ellps', 'WGS84')]
-        new_info['+a'] = ellps['a']
-        if 'b' not in ellps and 'rf' in ellps:
-            new_info['+f'] = 1. / ellps['rf']
+    
+    from pyproj import Geod
+    
+    
+    if 'ellps' in new_info:     
+        geod = Geod(**new_info)           
+        new_info['a'] = geod.a
+        new_info['b'] = geod.b      
+    else:
+        
+        if 'rf' in new_info and not 'f' in new_info:
+            new_info['f'] = 1. / float (new_info['rf']) 
+	       
+        if 'a' in new_info and 'b' in new_info:
```

Comment:
Could you move this condition to the `else` above? So it would be `if 'ellps' in new_info:` and `elif 'a' not in new_info or 'b' not in new_info`. Empty `pass` blocks rub me the wrong way.

Reply:
understood .. was just trying to make it obvious that if both a,b exist .. use those.  Changed - but not 100% sure it is what you were after.